lib/kubernetes-deploy/kubernetes_resource/replica_set.rb

```diff
@@ -40,6 +40,14 @@ def deploy_failed?
       @pods.present? && @pods.all?(&:deploy_failed?)
     end
 
+    def failure_message
+      @pods.map(&:failure_message).compact.uniq.join("\n")
```

Comment:
Not really. It's true that even for a large deployment that somehow manages to fail in multiple ways, in most cases (except maxUnavail/Surge 100%) the number of failed pods would be limited by the rollout strategy. I don't have any evidence that the number of comparisons likely to be involved would actually be slow.